client/client.go

```diff
@@ -112,35 +118,60 @@ func createClientWithClient(ctx context.Context, searchCli sserver1.SearchServer
 		return nil, err
 	}
 
-	indexer := libsearch.CreateSecureIndexBuilder(sha256.New, masterSecret, salts, uint64(size))
+	directoryInfos := make(map[string]DirectoryInfo)
 
-	var pathnameKey [32]byte
-	copy(pathnameKey[:], masterSecret[0:32])
+	// Initializes the info for each directory.
+	for i, directory := range directories {
+		indexer := libsearch.CreateSecureIndexBuilder(sha256.New, masterSecrets[i], salts, uint64(size))
 
-	absDir, err := filepath.Abs(directory)
-	if err != nil {
-		return nil, err
+		var pathnameKey [32]byte
+		copy(pathnameKey[:], masterSecrets[i][0:32])
+
+		absDir, err := filepath.Abs(directory)
+		if err != nil {
+			return nil, err
+		}
+
+		tlfID, err := getTlfID(absDir)
+		if err != nil {
+			return nil, err
+		}
+
+		dirInfo := DirectoryInfo{
+			tlfID:       tlfID,
+			indexer:     indexer,
+			pathnameKey: pathnameKey,
+		}
+
+		directoryInfos[absDir] = dirInfo
 	}
 
 	cli := &Client{
-		searchCli:   searchCli,
-		directory:   absDir,
-		indexer:     indexer,
-		pathnameKey: pathnameKey,
+		searchCli:      searchCli,
+		directoryInfos: directoryInfos,
 	}
 
 	return cli, nil
 }
 
-// AddFile indexes a file with the given `pathname` and writes the index to the
-// server.
-func (c *Client) AddFile(pathname string) error {
-	relPath, err := relPathStrict(c.directory, pathname)
+// AddFile indexes a file in `directory` with the given `pathname` and writes
+// the index to the server.
+func (c *Client) AddFile(directory, pathname string) error {
+	absDir, err := filepath.Abs(directory)
+	if err != nil {
+		return err
+	}
+
+	if _, ok := c.directoryInfos[absDir]; !ok {
```

Comment:
Actually I guess it's readonly.  So no need for locking for now.  Later we might want to add directories during runtime though, and then we'll need a lock.